docs/source/newsfragments/4208.change.rst

```diff
@@ -0,0 +1 @@
+Attempting to await on either the :class:`~cocotb.triggers.ReadWrite` or :class:`~cocotb.triggers.ReadOnly` trigger while in the ReadOnly state now raises a :exc:`RuntimeError`.
```

Comment:
```suggestion
Attempting to await on either the :class:`~cocotb.triggers.ReadWrite` or :class:`~cocotb.triggers.ReadOnly` trigger while in the ReadOnly phase now raises a :exc:`RuntimeError`.
```
We are calling this "phase" now in most places (https://docs.cocotb.org/en/latest/library_reference.html)